mempool/reactor.go

```diff
@@ -112,19 +112,21 @@ func (memR *MempoolReactor) broadcastTxRoutine(peer p2p.Peer) {
 
 	var next *clist.CElement
 	for {
+		select {
```

Comment:
We should probably preserve the old select and just add

```go
if !memR.IsRunning() || !peer.IsRunning() {
	return
}
```

here. What do you think?